File: Backend/src/main/java/edu/icet/dto/Patient.java
package edu.icet.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Patient {
    private int patientId;
    private String name;
    private String gender;
    private String phone;
    private String dob;
    private String address;
    private String bloodGroup;
    private String medicalHistory;
    private String allergies;
    private int userId;
}

File: Frontend/src/app/model/patient.ts
export class Patient {
    patientId: number;
    name: string;
    gender: string;
    phone: string;
    dob: string;
    address: string;
    bloodGroup: string;
    medicalHistory: string;
    allergies: string;
    userId: number;

    constructor(patientId: number, name: string, gender: string, phone: string, dob: string, address: string, bloodGroup: string, medicalHistory: string, allergies: string, userId: number) {
        this.patientId = patientId;
        this.name = name;
        this.gender = gender;
        this.phone = phone;
        this.dob = dob;
        this.address = address;
        this.bloodGroup = bloodGroup;
        this.medicalHistory = medicalHistory;
        this.allergies = allergies;
        this.userId = userId;
    }
}
